cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillController.java

```diff
@@ -118,20 +119,22 @@ public void release() {
   public void complete() {
     terminateOnServiceStop = true;
     executionService.stop();
-
     try {
+      RuntimeJobStatus status;
       RetryStrategy retryStrategy = RetryStrategies.timeLimit(
         5, TimeUnit.SECONDS, RetryStrategies.exponentialDelay(500, 2000, TimeUnit.MILLISECONDS));
 
       // Make sure the remote execution is completed
       // Give 5 seconds for the remote process to shutdown. After 5 seconds, issues a kill.
       long startTime = System.currentTimeMillis();
-      while (Retries.callWithRetries(remoteProcessController::isRunning, retryStrategy, Exception.class::isInstance)) {
+      while ((status = Retries.callWithRetries(
+        remoteProcessController::getStatus, retryStrategy, Exception.class::isInstance)) == RuntimeJobStatus.RUNNING) {
```

Comment:
How is it different than calling `isRunning`?

Reply:
```
  @Override
  public boolean isRunning() throws Exception {
    try (RuntimeJobManager runtimeJobManager = runtimeJobManagerSupplier.get()) {
      return !runtimeJobManager.getDetail(programRunInfo)
        .map(RuntimeJobDetail::getStatus)
        .map(RuntimeJobStatus::isTerminated)
        .orElse(true);
    }
  }

  @Override
  public RuntimeJobStatus getStatus() throws Exception {
    try (RuntimeJobManager runtimeJobManager = runtimeJobManagerSupplier.get()) {
      return runtimeJobManager.getDetail(programRunInfo)
        .map(RuntimeJobDetail::getStatus)
        .orElse(RuntimeJobStatus.UNKNOWN);
    }
  }

```

The functionality is the same effectively, we are planning to make use of the status in the refactored approach. In the form of 

```
      remoteProcessController.kill(status);

```